lib/improver/spotdata/extract_data.py

```diff
@@ -157,52 +159,111 @@ def process(self, cube, sites, neighbours, ancillary_data, additional_data,
-        A single dimensional coordinate is created using the running order,
-        whilst the non-monotonically increasing coordinates (e.g. bestdata_id)
-        are stored in AuxilliaryCoordinates.
+        Construct an iris.coord.Dim/Auxcoord using the provided options.
 
         Args:
         -----
-        latitudes : list
-            A list of latitudes ordered to match the sites OrderedDict.
+        data : number/list/np.array
+            List or array of values to populate the coordinate points.
+        coordinate : str
+            Name of the coordinate to be built.
+        coord_type : iris.coord.AuxCoord or iris.coord.DimCoord (optional)
+            Selection between Dim and Aux coord.
+        data_type : <type> (optional)
+            The data type of the coordinate points, e.g. int
+        units : str (optional)
+            String defining the coordinate units.
+        bounds : np.array (optional)
+            A (len(data), 2) array that defines coordinate bounds.
+        coord_system: iris.coord_systems.<coord_system> (optional)
+            A coordinate system in which the dimension coordinates are defined.
+        custom_function : function (optional)
+            A function to apply to the data values before constructing the
+            coordinate, e.g. np.nan_to_num.
 
-        longitudes : list
-            A list of longitudes ordered to match the sites OrderedDict.
+        Returns:
+        --------
+        iris coodinate : Dim or Auxcoord as chosen.
+
+        """
+        data = np.array(data, data_type)
+        if custom_function is not None:
+            data = custom_function(data)
+
+        crd_out = coord_type(data, long_name=coordinate, units=units,
+                             coord_system=coord_system, bounds=bounds)
+        crd_out.rename(coordinate)
+        return crd_out
+
+    @staticmethod
+    def _aux_coords_to_make():
+        """
+        Define coordinates that need to made for the cube to be produced.
 
-        altitudes : list
-            A list of altitudes ordered to match the sites OrderedDict.
+        """
+        return {'latitude': {'units': 'degrees', 'data_type': float,
+                             'coord_type': AuxCoord},
+                'longitude': {'units': 'degrees', 'data_type': float,
+                              'coord_type': AuxCoord},
+                'altitude': {'units': 'm', 'data_type': float,
+                             'coord_type': AuxCoord,
+                             'custom_function': np.nan_to_num},
+                'wmo_site': {'data_type': int, 'coord_type': AuxCoord},
+                'utc_offset': {'units': 'hours', 'data_type': float,
+                               'coord_type': AuxCoord}
+                }
 
-        utc_offsets : list
-            A list of UTC off sets in hours ordered to match the sites
-            OrderedDict.
+    @staticmethod
+    def make_stat_coordinate_first(cube):
```

Comment:
I've created #237, so that we can pull this functionality out into a top-level utility, and so that we can think about the generic functionality that we will want for multiple plugins.